jol-core/pom.xml

```diff
@@ -316,6 +316,89 @@ questions.
         </plugins>
     </build>
 
+    <profiles>
+        <profile>
+            <id>zgc-tests</id>
+            <activation>
+                <jdk>[17,)</jdk>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-surefire-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <id>test-zgc</id>
+                                <phase>test</phase>
+                                <goals>
+                                    <goal>test</goal>
+                                </goals>
+                                <configuration>
+                                    <argLine>-XX:+UseZGC</argLine>
+                                    <forkCount>4</forkCount>
+                                    <reuseForks>true</reuseForks>
+                                </configuration>
+                            </execution>
+                            <execution>
+                                <id>test-zgc-align16</id>
+                                <phase>test</phase>
+                                <goals>
+                                    <goal>test</goal>
+                                </goals>
+                                <configuration>
+                                    <argLine>-XX:+UseZGC -XX:ObjectAlignmentInBytes=16</argLine>
+                                    <forkCount>4</forkCount>
+                                    <reuseForks>true</reuseForks>
+                                </configuration>
+                            </execution>
+                        </executions>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+        <profile>
+            <id>generational-zgc-tests</id>
+            <activation>
+                <jdk>21</jdk>
```

Comment:
Sounds more like `[21,)`?

Reply:
Java 24+ doesn't have -XX:+ZGenerational flag and has non-generational mode removed. This profile forces usage of generational ZGC in current LTS version, the other profile enables default ZGC available.